elfi/examples/bignk.py

```diff
@@ -0,0 +1,296 @@
+    (2009): 189-201.
+
+    Parameters
+    ----------
+    a1 : float
+        The location (the 1st dimension).
+    a2 : float
+        The location (the 2nd dimension).
+    b1 : float
+        The scale (the 1st dimension).
+    b2 : float
+        The scale (the 2nd dimension).
+    g1 : float
+        The skewness (the 1st dimension).
+    g2 : float
+        The skewness (the 2nd dimension).
+    k1 : float
+        The kurtosis (the 1st dimension).
+    k2 : float
+        The kurtosis (the 2nd dimension).
+    rho : float
+        The dependence between components (dimensions), [1].
+    c : float, optional
+        The overall asymmetry parameter, as a convention fixed to 0.8 [2].
+    n_obs : int, optional
+        The number of the observed points
+    batch_size : int, optional
+    random_state : np.random.RandomState, optional
+
+    Returns
+    -------
+    array_like
+        The yielded points.
+    """
+
+    # Standardising the parameters
+    a1 = np.asanyarray(a1).reshape((-1, 1))
+    a2 = np.asanyarray(a2).reshape((-1, 1))
+    b1 = np.asanyarray(b1).reshape((-1, 1))
+    b2 = np.asanyarray(b2).reshape((-1, 1))
+    g1 = np.asanyarray(g1).reshape((-1, 1))
+    g2 = np.asanyarray(g2).reshape((-1, 1))
+    k1 = np.asanyarray(k1).reshape((-1, 1, 1))
+    k2 = np.asanyarray(k2).reshape((-1, 1, 1))
+    rho = np.asanyarray(rho).reshape((-1, 1))
+    A = np.hstack((a1, a2))
+    B = np.hstack((b1, b2))
+    g = np.hstack((g1, g2))
+    k = np.hstack((k1, k2))
+
+    # Sampling from the z term, Equation 3 [1].
+    z = []
+    for rho_i in rho:
+        matrix_cov = np.array([[1, rho_i], [rho_i, 1]])
+        z_el = ss.multivariate_normal.rvs(cov=matrix_cov, size=(n_obs),
+                                          random_state=random_state)
+        z.append(z_el)
+    z = np.array(z)
+
+    # Obtaining the first bracket term, Equation 3 [1].
+    gdotz = np.einsum('ik,ijk->ijk', g, z)
+    term_exp = (1 - np.exp(-gdotz)) / (1 + np.exp(-gdotz))
+    term_first = np.einsum('ik,ijk->ijk', B, (1 + c * (term_exp)))
+
+    # Obtaining the second bracket term, Equation 3 [1].
+    term_second_unraised = 1 + np.power(z, 2)
+    k = np.repeat(k, n_obs, axis=2)
+    k = np.swapaxes(k, 1, 2)
+    term_second = np.power(term_second_unraised, k)
+
+    # Yielding Equation 3, [1].
+    term_product = term_first * term_second * z
+    term_product_misaligned = np.swapaxes(term_product, 1, 0)
+    y_misaligned = np.add(A, term_product_misaligned)
+    y = np.swapaxes(y_misaligned, 1, 0)
+
+    return y
+
+
+def get_model(n_obs=150, true_params=None, stats_summary=None, seed_obs=None):
+    """Returns an initialised bivariate g-and-k model.
+
+    Parameters
+    ----------
+    n_obs : int, optional
+        The number of the observed points.
+    true_params : array_like, optional
+        The parameters defining the model.
+    stats_summary : array_like, optional
+        The chosen summary statistics, expressed as a list of strings.
```

Comment:
Add a list of options here, otherwise they are not visible without reading code.